src/devices/bus/snes/profighter.cpp

```diff
@@ -0,0 +1,254 @@
+	ROM_LOAD( "ami18cv8b.u16",  0x0e00, 0x0155, CRC(481a6974) SHA1(aff914837a7d8ef804b0ac4049d6f23ab23a1cda) )
+	ROM_LOAD( "palce16v8h.u17", 0x1000, 0x0117, CRC(70346ef5) SHA1(d9c0a7a2c466fa1fcc1680ee5dc3777714c05fe5) )
+	ROM_LOAD( "palce16v8h.u18", 0x1200, 0x0117, CRC(a3bbb6b1) SHA1(bdf291d2e3601d7b48c348b1d99b8872b1b4fedd) )
+ROM_END
+
+
+void sns_pro_fighter_qa_device::device_start()
+{
+}
+
+void sns_pro_fighter_qa_device::device_reset()
+{
+
+}
+
+const tiny_rom_entry *sns_pro_fighter_qa_device::device_rom_region() const
+{
+	return ROM_NAME( profghtqa );
+}
+
+void sns_pro_fighter_qa_device::device_add_mconfig(machine_config &config)
+{
+	WD37C65C(config, "fdc", 16_MHz_XTAL); // actually GM82C765B
+	FLOPPY_CONNECTOR(config, "fdc:0", profght_floppies, "35dd", floppy_image_device::default_pc_floppy_formats).enable_sound(true);
+}
+
+
+/*
+Super Pro. Fighter Q copier (variant 3)
+
+Main components:
+
+ACCMicro 3201 floppy disk controller
+24 MHz XTAL (for the FDC)
+unmarked CIC replacement
+D43256AGU-15LL CMOS Static RAM
+4 x GM71C4400BJ70 + 4 x KM44C1000BLJ-6 Fast Page DRAM (on a sub board)
+*/
+
+sns_pro_fighter_qb_device::sns_pro_fighter_qb_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, SNS_PRO_FIGHTER_QB, tag, owner, clock)
+	, device_sns_cart_interface(mconfig, *this)
+{
+}
+
+ROM_START( profghtqb ) // same program as profghtrq but different plds
+	ROM_REGION( 0x4000, "program", 0 )
+	ROM_LOAD( "u7", 0x0000, 0x4000, CRC(4f27e1fe) SHA1(ee11ac36b5e51476ab92c66135a2353e038c014b) )
+
+	ROM_REGION( 0xe00, "plds", ROMREGION_ERASE00 )
+	ROM_LOAD( "peel18cv8p.u2",  0x000, 0x155, CRC(e40bc5c0) SHA1(207437a41ae04bf4d5093775a6f2e602d0840d12) )
+	ROM_LOAD( "peel18cv8p.u4",  0x200, 0x155, CRC(bd152ad4) SHA1(bc79994d63efa7ef6e3cea4bd6f7940b68c40176) )
+	ROM_LOAD( "peel18cv8p.u5",  0x400, 0x155, CRC(b03c21a0) SHA1(dde7644cca7b383ab19e3598f30250a6695816fd) )
+	ROM_LOAD( "gal16v8b.u9",    0x600, 0x117, CRC(c9d37428) SHA1(9c06739d34a25095c0b0420910b82252ab0d05f8) )
+	ROM_LOAD( "peel18cv8p.u10", 0x800, 0x155, CRC(e0a3a486) SHA1(20162b8c174e73fb7d57d9a221eef87fb88005a5) )
+	ROM_LOAD( "gal16v8b.u11",   0xa00, 0x117, CRC(0b8a2c8f) SHA1(0cc9c9ed671f85a3ad8622b65f94b546bb13e6b6) )
+	ROM_LOAD( "gal16v8b.u12",   0xc00, 0x117, CRC(a1d23a1a) SHA1(d83dc1b50771f5d8b68293c85b0b94e257521bd3) )
+ROM_END
+
+
+void sns_pro_fighter_qb_device::device_start()
+{
+}
+
+void sns_pro_fighter_qb_device::device_reset()
+{
+
+}
+
+const tiny_rom_entry *sns_pro_fighter_qb_device::device_rom_region() const
+{
+	return ROM_NAME( profghtqb );
+}
+
+void sns_pro_fighter_qb_device::device_add_mconfig(machine_config &config)
+{
+	N82077AA(config, "fdc", 16_MHz_XTAL); // actually ACCMicro 3201
+	FLOPPY_CONNECTOR(config, "fdc:0", profght_floppies, "35dd", floppy_image_device::default_pc_floppy_formats).enable_sound(true);
+}
```

Comment:
You should probably add the cartridge slots to these devices – they have a SNES cartridge slot for connecting a game to copy, or a special cartridge providing a DSP chip for playing Pilot Wings, Mario Kart, etc.